internal/elasticsearch/pipeline/processors.go

```diff
@@ -0,0 +1,194 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipeline
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"sort"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+type Processor struct {
+	Type        string `yaml:"-"`
+	Tag         string
+	Description string
+	Line        int `yaml:"-"`
+}
+
+func (p Resource) Processors() (procs []Processor, err error) {
+	switch p.Format {
+	case "yaml", "yml":
+		procs, err = ProcessorsFromYAMLPipeline(p.Content)
+	case "json":
+		procs, err = ProcessorsFromJSONPipeline(p.Content)
+	default:
+		return nil, errors.Errorf("unsupported pipeline Format: %s", p.Format)
+	}
+	return procs, errors.Wrapf(err, "failure processing %s pipeline '%s'", p.Format, p.FileName())
+}
+
+func ProcessorsFromYAMLPipeline(content []byte) (procs []Processor, err error) {
+	type pipeline struct {
+		Processors []yaml.Node
+	}
+	var p pipeline
+	if err = yaml.Unmarshal(content, &p); err != nil {
+		return nil, err
+	}
+	for idx, entry := range p.Processors {
+		if entry.Kind != yaml.MappingNode || len(entry.Content) != 2 {
+			return nil, errors.Errorf("processor#%d is not a single key map (kind:%v Content:%d)", idx, entry.Kind, len(entry.Content))
+		}
+		var proc Processor
+		if err := entry.Content[1].Decode(&proc); err != nil {
+			return nil, errors.Wrapf(err, "error decoding processor#%d configuration", idx)
+		}
+		if err := entry.Content[0].Decode(&proc.Type); err != nil {
+			return nil, errors.Wrapf(err, "error decoding processor#%d type", idx)
+		}
+		proc.Line = entry.Line
```

Comment:
Nit. Could the `Line` of the last `Content` be used as the last line of the processor so we have per-line coverage instead of per-processor coverage? (and Cobertura paints red/green the whole processor :slightly_smiling_face:)

Reply:
Makes sense, done.

Originally I decided against it because it will mark the inner processors of a compound processor as covered (inner processor of `foreach`, and `on_failure` processors), while ES doesn't provide coverage information for those.